### ISSUE TITLE
terminals/friends/retails status

### ISSUE THREAD

**Issue by manfer** [irontec/ivozprovider#289]
I don't know if it is a bug or it is by design but the status of terminals, friends and retails does not change. Once a device is registered it looks like it remains with the green icon no matter if it goes offline. And I mean offline for a long time.

Two days ago at arround 11:00 UTC we unplugged a device that was registered as retail. Yesterday at 15:00 UTC it was still shown in ivozprovider with the online status icon. That's more than 24 hours. Today it is shown offline but I'm almost sure some services were restarted.

Today a friend has gone offline because an electrical problem with a UPS. At least it is offline for an hour now. Most probable it is some hours now because the problem must had started at night. Ivozprovider shows the friend online.

The terminals if I'm not wrong works exactly the same.

So, can ivozprovider show if a device goes offline after it has reported to be online? And, how long it takes for the system to realize a device is offline?

**Comment by Kaian** [comment]
Hi @manfer 

The field value is determined by the table kam_users_location (see the [ghost field code](https://github.com/irontec/ivozprovider/blob/oasis/library/IvozProvider/Klear/Ghost/RegisterStatus.php)).

When an endpoint registers, after some time its status is dumped by kamailio@users into database. Then the portal reads that value. This value should be valid until the registration expires (based on Expire header of the SIP REGISTER message).

This value is never realtime. A good value for Expire could be, for example 1800 seconds. If the value is too small introduces too much overload when there are too many endpoints.

When Expire time has past or when the endpoint sent a REGISTER with the Expire header set to 0, the registration should end. The changes in database (what the ghost field shows) won't display this until kamailio@users dump those status to database, but it should not take more than a couple of minutes at worst.

Also note that friends and retails that have direct connectivity always display a green arrow no matter if they're alive or not.

Regards,

**Comment by cruzccl** [comment]
Hi @manfer ,

We have made some tests and this seems to be working as expected: when register expiration shown in tooltip arrives and the register has not been renewed, the icon gets red.

It is important to understand that there is no underlying mechanism that monitorizes UACs/friends (sending OPTIONS or similar) to keep this icon updated as soon as connectivity is lost: it keeps green until registration expires without a renewal.

By the way, I have just opened #290 to make some changes on registration minimal and maximum expire values.


Regards,
